server/boutique-hotel-technikum/src/main/java/com/example/boutiquehoteltechnikum/controllers/RoomController.java

```diff
@@ -17,29 +23,53 @@ public class RoomController {
     @Autowired
     private RoomService roomService;
 
+    @Autowired
+    private RoomTransformer roomTransformer;
+
     @GetMapping("")
-    public List<RoomDto> getRooms() {
-        List<RoomEntity> roomEntities = roomService.getRooms();
+    public RoomResponseObject getRooms(@PathParam("limit") int limit, @PathParam("offset") int offset) {
+        if (limit <= 0) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, String.format("Limit (%d) cannot be less or equal zero", limit));
+        }
+        if (offset < 0) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, String.format("Offset (%d) cannot be less than zero", offset));
+        }
+        Page<RoomEntity> roomPages = roomService.getRooms(limit, offset);
+        List<RoomEntity> roomEntities = roomPages.getContent();
         List<RoomDto> roomDtos = new ArrayList<>();
         for (RoomEntity roomEntity : roomEntities) {
-            roomDtos.add(
-                RoomDto.builder()
-                    .id(roomEntity.getId())
-                    .name(roomEntity.getName())
-                    .description(roomEntity.getDescription())
-                    .build());
+            roomDtos.add(roomTransformer.toDto(roomEntity));
         }
-        return roomDtos;
+        return RoomResponseObject.builder()
+            .limit(limit)
+            .offset(offset)
+            .total(roomPages.getTotalElements())
+            .roomDtos(roomDtos)
+            .build();
+    }
+
+    @GetMapping("/availability")
+    public RoomResponseObject getAvailableRooms(@RequestParam("startDate") String startDate, @RequestParam("endDate") String endDate) {
+        List<RoomEntity> availableRooms = roomService.getAvailableRooms(startDate, endDate);
+        List<RoomDto> roomDtos = new ArrayList<>();
+        for (RoomEntity roomEntity : availableRooms) {
+            roomDtos.add(roomTransformer.toDto(roomEntity));
+        }
+        return RoomResponseObject.builder()
+            .limit(roomDtos.size())
+            .offset(0)
+            .total(roomDtos.size())
+            .roomDtos(roomDtos)
+            .build();
```

Comment:
Please combine these two like `getRoomById` and if no start or end date is specified, all Rooms are loaded but if many rooms are available, they are returned with limit and offset (Important: total is then only the number of available rooms in the specified time range)